### ISSUE TITLE
 Document the JS component tree #3846 

### ISSUE THREAD

**Issue by Adiman007** [grandma-collaboration/skyportal_grandma_dev#45]
Ok, I open an issue here regarding this issue ( https://github.com/skyportal/skyportal/issues/3846 )
@KodeurFou and I will have a look on if something already exist to make an automated JSX component tree and if not we will try to make a script that check all the files in the component directory and then check the dependencies. 
We will go back to you when we have something that might be useful :+1: 
@mcoughlin @Theodlz

**Comment by Adiman007** [comment]
hi @Theodlz, quick question, 
Guy said "Add a short code comment at the beginning of each file, saying e.g., "this is in the Source page, under the annotations table, etc."." 
So are we supposed to add a comment on the 250 components of the app ?

**Comment by mcoughlin** [comment]
@Adiman007 No. I foresee this as part of the docs, i.e. some kind of figure and how to understand it.

**Comment by Theodlz** [comment]
Personally what I got from Guy's comment is to add a short comment (few lines) at the top of each file to describe/explain what it contains, and NOT add comments for every methods.

**Comment by Theodlz** [comment]
You just need to read each file quickly, understand what each component does and then add a short summary of what you understood at the top of the file.

That should also help you guys a lot to understand the entire frontend architecture of SP.

**Comment by KodeurFou** [comment]
![dependency-graph](https://user-images.githubusercontent.com/104773256/219322258-2d0c106d-4014-4882-b16a-039654794e80.svg)

We used dependency-cruiser to model the architecture of SP, we can see the react components and their relations

The graph is a little messy we tried having it displayed in a reactive way in an html format but so far no success

The problem we've had is it doesn't seem to be fitted to run on the same version of react that SP does
![image](https://user-images.githubusercontent.com/104773256/219324448-a0526de1-0333-4bb1-818e-38d88a7404d1.png)

but the output we still get seems to be correct

**Comment by KodeurFou** [comment]
should we try another way to try and automate the process or should we document the way we got it working so it can be rerendered when some new components are added?

**Comment by mcoughlin** [comment]
@KodeurFou @Adiman007 Could you open this as a PR or issue on skyportal to discuss?

**Comment by KodeurFou** [comment]
I've copied it to the original issue on skyportal !
